app/scripts/modules/core/src/application/nav/ApplicationNavigation.tsx

```diff
@@ -50,32 +58,56 @@ export const ApplicationNavigation = ({ app }: IApplicationNavigationProps) => {
     };
   }, []);
 
+  React.useEffect(() => {
```

Comment:
The global state of the cache changes when the toggle button is clicked, however this component will not re-render unless I subscribe to the cache. 

I think context would probably work. They are not in the same component tree, but I can use the cache value and provide that context to the application navigation component

Reply:
>They are not in the same component tree

recoiljs looks very useful for cases like this

Reply:
Just looked into context and recoiljs. Both have challenges. 

Recoil requires the provider to be at the root of that app. React context requires the provider to be above both components so they are in the same tree (which is the root of the app in this case). This is currently in angular and I am not sure how ready it is for a refactor. I will continue digging into a workaround - a partial refactor of `spinnaker.component` could work. I just need the container contents to be in react

Reply:
> This is currently in angular and I am not sure how ready it is for a refactor

ah bummer. yeah, migrating that `spinnaker.component` would be a win.  The one sticky point will be handling the spinnakerHeaderTemplate include.

fwiw, recoil is still in incubation, but it looks pretty slick.

Reply:
🤔 actually, this might be really easy.

Since `SpinnakerHeader` is @Overridable I think it wires in the angularjs template override automatically.  It might be as easy as replacing the `ng-include` with the `<SpinnakerHeader />`.  You can check deck-nflx to see if our custom header gets rendered.

The rest of those components used in that template are already downgraded React -> AngularJS components so this might be quick and easy

Reply:
... and to continue talking to my self, I see you have a PR out there already to migrate that component :) #8366

Reply:
Recoil looks pretty lightweight, so I got excited to try it out. I think we can make this happen, the overridable is the thing I was missing, but the rest was very easy to migrate. Thanks for all the input